skglm/experimental/reweighted.py

```diff
@@ -0,0 +1,107 @@
+import numpy as np
+from numpy.linalg import norm
+from skglm.estimators import Lasso
+
+
+def _reweight_by_coefficient_norm(coef):
+    nrm = np.sqrt(norm(coef))
+    return 1 / (2 * nrm + np.finfo(float).eps)
+
+
+class ReweightedLasso(Lasso):
+    r"""Reweighted Lasso estimator.
```

Comment:
Let's go for a general scheme, where the objective is not explicitely L05. We just solve a concave penalty by a MM/DC approach.